storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java

```diff
@@ -0,0 +1,518 @@
+
+        public void validateField(String name, Class[] keyValidators, Class[] valueValidators, Object o) throws IllegalAccessException, InstantiationException {
+            if (o == null) {
+                return;
+            }
+            //check if Map
+            SimpleTypeValidator isMap = new SimpleTypeValidator();
+            isMap.validateField(name, Map.class, o);
+            for (Map.Entry<Object, Object> entry : ((Map<Object, Object>) o).entrySet()) {
+                for (Class kv : keyValidators) {
+                    Object keyValidator = kv.newInstance();
+                    if (keyValidator instanceof Validator) {
+                        ((Validator) keyValidator).validateField(name + " Map key", entry.getKey());
+                    } else {
+                        LOG.warn("validator: {} cannot be used in MapEntryCustomValidator to validate keys.  Individual entry validators must a instance of Validator class", kv.getName());
+                    }
+                }
+                for (Class vv : valueValidators) {
+                    Object valueValidator = vv.newInstance();
+                    if (valueValidator instanceof Validator) {
+                        ((Validator) valueValidator).validateField(name + " Map value", entry.getValue());
+                    } else {
+                        LOG.warn("validator: {} cannot be used in MapEntryCustomValidator to validate values.  Individual entry validators must a instance of Validator class", vv.getName());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Validates a positive number
+     */
+    public static class PositiveNumberValidator extends Validator{
+
+        @Override
+        public void validateField(String name, Object o) {
+            validateField(name, false, o);
+        }
+
+            public void validateField(String name, boolean includeZero, Object o) {
+            if (o == null) {
+                return;
+            }
+            if (o instanceof Number) {
+                if(includeZero == true) {
+                    if (((Number) o).doubleValue() >= 0.0) {
+                        return;
+                    }
+                } else {
+                    if (((Number) o).doubleValue() > 0.0) {
+                        return;
+                    }
+                }
+            }
+            throw new IllegalArgumentException("Field " + name + " must be a Positive Number");
+        }
+    }
+
+    /**
+     * Methods for validating confs
+     */
+
+    /**
+     * Validates a field given field name as string
+     *
+     * @param fieldName provided as a string
+     * @param conf      map of confs
+     */
+    public static void validateField(String fieldName, Map conf) throws NoSuchFieldException, IllegalAccessException, InstantiationException, NoSuchMethodException, InvocationTargetException {
+        validateField(fieldName, conf, CONFIG_CLASS);
+    }
```

Comment:
useful if a developer would like to validate a field in a conf map at any point.  The developer can just call:

 validateField(Config.EXAMPLE_CONF, confMap)

instead of trying to get the corresponding Field object first.  Just make the code cleaner if someone else decides to validates some configs in the future